lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/main.js

```diff
@@ -0,0 +1,83 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var logger = require( 'debug' );
+var parser = require( '@conventional-commits/parser' ).parser;
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var format = require( '@stdlib/string/format' );
+var merge = require( '@stdlib/utils/merge' );
+var toConventionalChangelog = require( './conventional_changelog.js' );
+var extractCommits = require( './commits.js' );
+
+
+// VARIABLES //
+
+var debug = logger( 'changelog:parse-commits' );
+
+
+// MAIN //
+
+/**
+* Parses package commit messages into conventional changelog objects.
+*
+* @param {string} pkg - package name
+* @throws {TypeError} must provide a string
+* @returns {Array} array of conventional changelog formatted commit message objects
+*
+* @example
+* var out = parseCommits( 'utils/curry' );
+* // returns [...]
+*/
+function parseCommits( pkg ) {
+	var commits;
+	var merged;
+	var msg;
+	var ast;
+	var out;
+	var i;
+	var j;
+
+	if ( !isString( pkg ) ) {
+		throw new TypeError( format( 'invalid argument. Must provide a string. Value: `%s`.', pkg ) );
+	}
+	commits = extractCommits( pkg );
+	out = [];
+	for ( i = 0; i < commits.length; i++ ) {
+		try {
+			msg = commits[ i ].message;
+			ast = parser( msg );
+			merged = merge( commits[ i ], toConventionalChangelog( ast ) );
+			for ( j = 0; j < merged.notes.length; j++ ) {
+				merged.notes[ j ].hash = commits[ i ].hash;
+			}
+			out.push( merged );
```

Comment:
You have multiple lines in this `try` block. It would be good to extract out only those lines for which we believe are allowed to throw. Otherwise, you are swallowing and conflating syntax errors with runtime errors.

Reply:
Addressed.